Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -423,3 +423,28 @@ theorem tensorProduct_mapIncl_injective_of_left
 end Flat
 
 end Module
+
+section Injective
+
+open scoped TensorProduct
+
+variable {R S A B : Type*} [CommRing R] [Ring A] [Algebra R A] [Ring B] [Algebra R B]
+  [CommSemiring S] [Algebra S A] [SMulCommClass R S A]
+
+namespace Algebra.TensorProduct
+
+theorem includeLeft_injective [Module.Flat R A] (hb : Function.Injective (algebraMap R B)) :
+    Function.Injective (includeLeft : A →ₐ[S] A ⊗[R] B) := by
+  convert Module.Flat.lTensor_preserves_injective_linearMap (M := A) (Algebra.linearMap R B) hb
+    |>.comp (_root_.TensorProduct.rid R A).symm.injective
```

Comment:
Do you need the `_root_`? Apparently not even if the `Algebra` namespace is open, because both `rid` are protected:
```
import Mathlib
open Algebra.TensorProduct
#check TensorProduct.rid
```

Reply:
Seems that inside the `Algebra.TensorProduct` namespace, `TensorProduct.rid` refers to `Algebra.TensorProduct.rid` and it errors...

Reply:
Indeed, I can reproduce.